library/Fission/Web.hs

```diff
@@ -7,106 +7,108 @@ module Fission.Web
   , server
   ) where
 
-import           Flow
-import           RIO
-
-import           Data.Has
-import           Database.Selda
 import           Servant
 
-import qualified Fission.Config     as Config
-import           Fission.User
-
+import qualified Network.AWS.Auth   as AWS
 import           Network.IPFS
 import qualified Network.IPFS.Types as IPFS
 
-import           Fission.Internal.Orphanage.PlainText ()
+import           Fission.Prelude
+import qualified Fission.Config     as Config
+
 import           Fission.Internal.Orphanage.OctetStream ()
+import           Fission.Internal.Orphanage.PlainText   ()
 
-import qualified Fission.Web.Auth        as Auth
-import qualified Fission.Web.DNS         as DNS
-import qualified Fission.Web.IPFS        as IPFS
-import qualified Fission.Web.Ping        as Ping
-import qualified Fission.Web.Routes      as Web
-import           Fission.Web.Server
-import qualified Fission.Web.Swagger     as Web.Swagger
-import qualified Fission.Web.Types       as Web
-import qualified Fission.Web.User        as User
+import qualified Fission.Platform.Heroku.ID.Types       as Heroku
+import qualified Fission.Platform.Heroku.Password.Types as Heroku
 
-import qualified Network.AWS.Auth  as AWS
-import qualified Fission.AWS.Types as AWS
+import qualified Fission.AWS.Types   as AWS
 
-import qualified Fission.Platform.Heroku.Types as Heroku
-import qualified Fission.Web.Heroku            as Heroku
+import           Fission.Web.Server
+import qualified Fission.Web.Auth    as Auth
+import qualified Fission.Web.DNS     as DNS
+import qualified Fission.Web.Heroku  as Heroku
+import qualified Fission.Web.IPFS    as IPFS
+import qualified Fission.Web.Ping    as Ping
+import qualified Fission.Web.Routes  as Web
+import qualified Fission.Web.Swagger as Web.Swagger
+import qualified Fission.Web.Types   as Web
+import qualified Fission.Web.User    as User
 
 -- | Top level web API type. Handled by 'server'.
 type API = Web.Swagger.API :<|> Web.API
 
 app ::
   ( MonadLocalIPFS  (RIO cfg)
   , MonadRemoteIPFS (RIO cfg)
-  , Has IPFS.Gateway     cfg
-  , Has IPFS.Peer        cfg
-  , Has Web.Host         cfg
-  , Has AWS.AccessKey    cfg
-  , Has AWS.SecretKey    cfg
-  , Has AWS.ZoneID       cfg
-  , Has AWS.DomainName   cfg
-  , Has Heroku.ID        cfg
-  , Has Heroku.Password  cfg
-  , HasLogFunc           cfg
-  , MonadSelda      (RIO cfg)
+  , MonadTime       (RIO cfg)
+  , MonadDB         (RIO cfg)
+  , MonadDB                 m
+  , MonadReader         cfg m
+  , Has IPFS.Gateway    cfg
+  , Has IPFS.Peer       cfg
+  , Has Web.Host        cfg
+  , Has AWS.AccessKey   cfg
+  , Has AWS.SecretKey   cfg
+  , Has AWS.ZoneID      cfg
+  , Has AWS.DomainName  cfg
+  , Has Heroku.ID       cfg
+  , Has Heroku.Password cfg
+  , HasLogFunc          cfg
```

Comment:
This list is currently _longer_. What the heck, right? Well, currently this function is contending with needing to describe a concrete `RIO` _and_ abstract capabilities like `MonadTime`. Hence some oddities like `MonadTime (RIO cfg)`.